### ISSUE TITLE
Grid - wrong row/index goes in edit-mode after applying filter

### ISSUE THREAD

**Issue by unrealmaker** [telerik/kendo-angular#1450]
## I'm submitting a...

* Bug report 

## Current behavior
When trying to edit a specific row after the grid was filtered, the edit mask is shown on a different row or not shown at all.

## Expected behavior
When editing a specific row the edit mask should be shown on that row.

## Minimal reproduction of the problem with instructions

Here a demo: https://plnkr.co/edit/zn2twFfY9Y7av9adBi1l?p=preview (taken from https://www.telerik.com/kendo-angular-ui/components/grid/editing/editing-reactive-forms/ and enabled filtering).

Steps to reproduce:
1. Type "u" in the filter for "ProductName"
2. Click on the "edit" button of the first row
3. The third row will go in "edit-mode"

## What is the motivation or use case for changing the behavior?
This data could get corrupted easily. The grid should edit the correct rows.

## Environment

Package versions:

```
@progress/kendo-angular-grid@2.1.3
@progress/kendo-angular-grid@3.0.0
```

Browser:

- Chrome (desktop) version 65.0.3325.181

System:
- TypeScript version: 1.3.0.0
- Node version: v6.10.3  (NPM: 3.10.10)
- Platform:  Windows

The bug appeared in v2.1.3.

**Comment by ipeshev** [comment]
Fixed in v3.1.0-dev.201804160839

**Comment by tsvetomir** [comment]
Published in v3.1.0
